File: 2155-find-missing-observations/find-missing-observations.py
class Solution(object):
    def missingRolls(self, rolls, mean, n):
        """
        :type rolls: List[int]
        :type mean: int
        :type n: int
        :rtype: List[int]
        """
        m=len(rolls)
        tot=(m+n)*mean
        cur=0
        for i in range(m):
            cur+=rolls[i]
        
        ans=[]
        d=(tot-cur)//n
        mod=(tot-cur)%n
        if d>=1 and d<=5 or d==6 and mod==0:
            for i in range(n):
                ans.append(d)
                if mod>=1:
                    ans[i]+=1
                    mod-=1
        return ans

File: 2155-find-missing-observations/find-missing-observations.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int m=rolls.size();
        int tot=(m+n)*mean;
        int cur=0;
        for(int i=0;i<m;i++){
            cur+=rolls[i];
        }
        vector <int> ans;
        int d=(tot-cur)/n;
        int mod=(tot-cur)%n;
        //cout << tot << "  " << cur << endl;
        if(d>=1 && d<=5 || d==6 && mod==0){
            for(int i=0;i<n;i++){
                ans.push_back(d);
                if(mod){
                    ans[i]++;
                    mod--;
                }
            }
        }
        return ans;
    }
};